rnd/autogpt_server/autogpt_server/data/model.py

```diff
@@ -0,0 +1,51 @@
+from typing import ClassVar, Optional
+
+from pydantic import Field
+from pydantic.fields import FieldInfo
+
+from autogpt_server.util.settings import Secrets
+
+
+class BlockSecret:
+    def __init__(self, key=None, value=None):
+        if value is not None:
+            self._value = value
+            return
+
+        self._value = self.__get_secret(key)
+        if self._value is None:
+            raise ValueError(f"Secret {key} not found.")
+
+    STR: ClassVar[str] = "<secret>"
+    SECRETS: ClassVar[Secrets] = Secrets()
+
+    def __repr__(self):
+        return BlockSecret.STR
+
+    def __str__(self):
```

Comment:
So what the schema for this type looks like now?

Also, can the user still provide a string to override the secret and be parsed as a blocksecret?

Reply:
I've just brought back the `__get_pydantic*` methods for `BlockSecret` because it was broken without them.

> So what the schema for this type looks like now?

```
"type": "string",
"title": "BlockSecret",
```

> can the user still provide a string to override the secret and be parsed as a blocksecret

yes, I brought back `parse_value` as well

`BlockSecret` is a regular class that *can* be used anywhere (including as a model field). It behaves like previosely but doesn't provide description (because it cannot be specified for each object in the schema - schema is shared for all objects in this case).  
It's recommended to use `SecretField` function for model fields - it instantiates actual Pydantic field and that allows to specify description and title for each Field.